packages/neuron-ui/src/components/Network/index.tsx

```diff
@@ -27,28 +33,55 @@ const Span = styled.span`
   align-items: center;
 `
 
-const NetworkStatusHeader = () => {
+const NetworkStatusHeader = ({
+  networks,
+  navTo,
+  dispatch,
+}: {
+  networks: Network[]
+  dispatch: React.Dispatch<{ type: HeaderActions; payload?: any }>
+  navTo: Function
+  actionCreators: HeaderActionsCreators
+}) => {
   const chain = useContext(ChainContext)
-  const [t] = useTranslation()
+
+  const tipNumber = chain.tipBlockNumber === undefined ? undefined : `#${(+chain.tipBlockNumber).toLocaleString()}`
+  const networkItems = [
+    ...networks.map(network => ({
+      label: network.name || network.remote,
+      onClick: () => {
+        dispatch(actionCreators.setNetwork(network))
+      },
+    })),
+    {
+      label: 'Mangement',
+      onClick: () => navTo(Routes.SettingsNetworks),
+    },
+  ]
+
   return (
-    <FlexDiv>
-      <Span>
-        <Status
+    <>
+      <FlexDiv>
+        <Span>
+          <Status
+            style={{
+              color: chain.network.status === NetworkStatus.Online ? 'green' : 'red',
+            }}
+          />
+        </Span>
+        <Span>{`${(chain.network.name && chain.network.name.slice(0, 30)) || chain.network.remote} - `}</Span>
```

Comment:
Looks like this should go to `Network`. For example, add a `label` function which use `chain.network.name && chain.network.name.slice(0, 30)) || chain.network.remote`.